org.opentosca.container.api/src/org/opentosca/container/api/controller/PlacementController.java

```diff
@@ -47,36 +49,50 @@
     String serviceTemplateId;
 
     private final InstanceService instanceService;
+    private final NodeTemplateService nodeTemplateService;
 
-    public PlacementController(final InstanceService instanceService) {
+    public PlacementController(final InstanceService instanceService, final NodeTemplateService nodeTemplateService) {
         this.instanceService = instanceService;
+        this.nodeTemplateService = nodeTemplateService;
     }
 
     @POST
-    @Produces({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
-    @Consumes({MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML})
+    @Produces({MediaType.APPLICATION_JSON})
+    @Consumes({MediaType.APPLICATION_JSON})
     @ApiOperation(hidden = true, value = "")
-    public Response getInstances(@ApiParam("node template list need to be placed") final PlacementModel request) throws InstantiationException,
-                                                                                                                 IllegalAccessException,
-                                                                                                                 IllegalArgumentException {
+    public Response getInstances(@ApiParam("node template list need to be placed") final List<String> request) throws InstantiationException,
+                                                                                                               IllegalAccessException,
+                                                                                                               IllegalArgumentException {
+
         // all node templates that need to be placed
-        final List<PlacementNodeTemplate> nodeTemplatesToBePlaced = request.getNeedToBePlaced();
+        final List<String> nodeTemplateIdsToBePlaced = request;
+        final List<NodeTemplateDTO> nodeTemplatesToBePlaced = new ArrayList<>();
+
+        nodeTemplateIdsToBePlaced.stream()
+                                 .forEach(id -> nodeTemplatesToBePlaced.add(this.nodeTemplateService.getNodeTemplateById(this.csarId,
+                                                                                                                         QName.valueOf(this.serviceTemplateId),
+                                                                                                                         id)));
```

Comment:
this is probably better formulated as a chain using map like so:

```java
nodeTemplateIdsToBePlaced.stream()
  .map(id -> nodeTemplateService.getNodeTemplateById(csarId, QName.valueOf(serviceTemplateId), id))
  .forEach(nodeTemplatesToBePlaced::add)
```

This makes it clear that `nodeTemplatesToBePlaced` could as well be constructed using a collector `collect(Collectors.toList())` especially since the list is not modified hereafter